org.lflang/src/org/lflang/generator/ts/TSReactionGenerator.kt

```diff
@@ -142,6 +143,49 @@ class TSReactionGenerator(
         }
     }
 
+    private fun generateMutationString(
+        mutation: Mutation,
+        mutationPrologue: String,
+        mutationEpilogue: String,
+        mutationFuncArgs: StringJoiner,
+        mutationSignature: StringJoiner
+    ): String {
+        // Assemble reaction triggers
+        val mutationTriggers = StringJoiner(",\n")
+        for (trigger in mutation.triggers) {
+            if (trigger is VarRef) {
+                mutationTriggers.add(trigger.generateVarRef())
+            } else if (trigger is BuiltinTriggerRef) {
+                when (trigger.type) {
+                    BuiltinTrigger.STARTUP  -> mutationTriggers.add("this.startup")
+                    BuiltinTrigger.SHUTDOWN -> mutationTriggers.add("this.shutdown")
+                    else -> {}
+                }
+            }
+        }
```

Comment:
This part of generateMutationString() seems to be exactly the same as that of generateReactionString(). How about we factor this out as a separate function and have it reused in both generateMutationString() and generateReactionString()?

Reply:
https://github.com/lf-lang/lingua-franca/pull/1252/commits/8dfcc2a223a5648cbed67ad6b615079036c3d4b3
I combined the generateMutationString() and generateReactionString().
It works successfully, but I'm not sure about my logic. @hokeun